### ISSUE TITLE
TextEditor: Default value XSS

### ISSUE THREAD

**Issue by ravihariharan** [primefaces/primefaces#7484]
**Description**

**Scenario 1:**
 `<p:textEditor value="#{dtBasicView.xssStr}" /> where xssStr is blank`

Without owasp-java-html-sanitizer-20181114.1.jar (HTML Not Sanitized )
Enter `<script type="text/javascript">alert("1111");</script>`
Click Button - Logs `<p>&lt;script type="text/javascript"&gt;alert("1111");&lt;/script&gt;</p>`

With owasp-java-html-sanitizer-20181114.1.jar (HTML Sanitized )
Enter `<script type="text/javascript">alert("1111");</script>`
Click Button - Logs `<p>&lt;script type&#61;&#34;text/javascript&#34;&gt;alert(&#34;1111&#34;);&lt;/script&gt;</p><p><br /></p>`

**No issue so far :)** 

**Scenario 2:**
`<p:textEditor value="#{dtBasicView.xssStr}" /> where xssStr is "<script type="text/javascript">alert("1111");</script>`
On page load a alert popup is shown.

**Describe the solution you would like**

**Should primefaces have support to sanitize html for text editor on client side ?**
_https://github.com/quilljs/quill/issues/1918
the package sanitize-html is quite efficient for this_

**Additional context**
The issue in when we compare it with behavior with output label , we have to explicitly set escape=false which we know could lead to XSS issue.

**Example XHTML**
```xml
<h:form id="form">
        <p:textEditor value="#{dtBasicView.xssStr}" />
        <p:commandButton value="Submit" action="#{dtBasicView.save}" />
    </h:form>
```

**Example Bean**
```java
@Named("dtBasicView")
@ViewScoped
public class BasicView implements Serializable
{
    private String xssStr = "<script type="text/javascript">alert("1111");</script>";

    public String getXssStr()
    {
        return xssStr;
    }

    public void setXssStr(String xssStr)
    {
        this.xssStr = xssStr;
    }

    public void save()
    {
        System.out.println(this.xssStr);
    }
}
```

https://forum.primefaces.org/viewtopic.php?f=3&t=67514&p=192413#p192413

**Comment by ravihariharan** [comment]
@mertsincan / @sqores  can this fix be part of elite 7.0.23 ?

**Comment by buunsal** [comment]
@ravihariharan Ok, I included this fix for 7.0.23.

**Comment by ravihariharan** [comment]
Thanks @sqores :)
